odfe-cli/it/ad_test.go

```diff
@@ -0,0 +1,347 @@
+		Index:       a.DetectorRequest.Index,
+		Features: []adentity.Feature{
+			{
+				Name:             "sum_value",
+				Enabled:          true,
+				AggregationQuery: getRawFeatureAggregation(),
+			},
+		},
+		Interval: adentity.Interval{
+			Period: adentity.Period{
+				Duration: 1,
+				Unit:     "Minutes",
+			},
+		},
+		Delay: adentity.Interval{
+			Period: adentity.Period{
+				Duration: 1,
+				Unit:     "Minutes",
+			},
+		},
+	}
+}
+
+func (a *ADTestSuite) TearDownSuite() {
+	a.DeleteEcommerceIndex()
+}
+
+// This will run right before the test starts
+// and receives the suite and test names as input
+func (a *ADTestSuite) BeforeTest(suiteName, testName string) {
+	// We don't need to create detector for create use case
+	if testName != "TestCreateDetectors" {
+		a.CreateDetectorUsingRESTAPI(a.T())
+	}
+}
+
+// This will run after test finishes
+// and receives the suite and test names as input
+func (a *ADTestSuite) AfterTest(suiteName, testName string) {
+	if testName != "TestCreateDetectors" || a.DetectorId != "" {
+		a.StopDetectorUsingRESTAPI(a.T(), a.DetectorId)
+		a.DeleteDetectorUsingRESTAPI(a.T(), a.DetectorId)
+	}
+}
+
+//DeleteDetectorUsingRESTAPI helper to delete detector using rest api
+func (a *ADTestSuite) DeleteDetectorUsingRESTAPI(t *testing.T, ID string) {
+	indexURL := fmt.Sprintf("%s/_opendistro/_anomaly_detection/detectors/%s", a.Profile.Endpoint, ID)
+	_, err := a.callRequest(http.MethodDelete, []byte(""), indexURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+//StartDetectorUsingRESTAPI helper to start detector using rest api
+func (a *ADTestSuite) StartDetectorUsingRESTAPI(t *testing.T, ID string) {
+	if ID == "" {
+		t.Fatal("Detector ID cannot be empty")
+	}
+	indexURL := fmt.Sprintf("%s/_opendistro/_anomaly_detection/detectors/%s/_start", a.Profile.Endpoint, ID)
+	_, err := a.callRequest(http.MethodPost, []byte(""), indexURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+//StopDetectorUsingRESTAPI helper to stop detector using rest api
+func (a *ADTestSuite) StopDetectorUsingRESTAPI(t *testing.T, ID string) {
+	if ID == "" {
+		t.Fatal("Detector ID cannot be empty")
+	}
+	indexURL := fmt.Sprintf("%s/_opendistro/_anomaly_detection/detectors/%s/_stop", a.Profile.Endpoint, ID)
+	_, err := a.callRequest(http.MethodPost, []byte(""), indexURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+//CreateDetectorUsingRESTAPI helper to create detector using rest api
+func (a *ADTestSuite) CreateDetectorUsingRESTAPI(t *testing.T) {
+	indexURL := fmt.Sprintf("%s/_opendistro/_anomaly_detection/detectors", a.Profile.Endpoint)
+	reqBytes, err := json.Marshal(a.Detector)
+	if err != nil {
+		t.Fatal(err)
+	}
+	response, err := a.callRequest(http.MethodPost, reqBytes, indexURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var data map[string]interface{}
+	_ = json.Unmarshal(response, &data)
+	if val, ok := data["_id"]; ok {
+		a.DetectorId = fmt.Sprintf("%s", val)
+		return
+	}
+	t.Fatal(data)
+}
+
+func getRawFilter() []byte {
```

Comment:
Could this be formatted better? Also, could this be parametrized with field name and value?

Reply:
formatted now. It is not required since we are not parameterizing for different test cases.

Reply:
makes sense